File: src/128.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        set<int> numsSet;
        int maxStreak = 1, count = 1;
        for (int i = 0; i < nums.size(); i++) {
            numsSet.insert(nums[i]);
        }
        for (auto it = numsSet.begin(); it != numsSet.end(); it++) {
            count = 1;
            while (true) {
                if (it != numsSet.end() && numsSet.find(*it + 1) != numsSet.end()) {
                    it++;
                    count++;
                } else break;
            }
            maxStreak = max(maxStreak, count);
        }
        return maxStreak;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {100, 4, 200, 1, 3, 2};
    cout << solution.longestConsecutive(nums);
    return 0;
}

File: pysrc/128.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        s = set()
        for num in nums:
            s.add(num)

        longest = 1
        for num in s:
            if not num - 1 in s:
                i = num + 1
                while i in s:
                    i += 1
                longest = max(longest, i - num)
        return longest


if __name__ == '__main__':
    solution = Solution()
    print(solution.longestConsecutive([100, 4, 200, 1, 3, 2]))
